### ISSUE TITLE
Generate a trace file like this

### ISSUE THREAD

**Issue by liuhy397** [gpgpu-sim/gpgpu-sim_distribution#136]
I want generate a file like this,what should i do for gpgpu-sim,thank you very much.
![Uploading 193d0c00913cda9d3aa822b5064b6ae.jpg…]()

**Comment by gangmul12** [comment]
is it dram trace or fetch request generated at processor?

**Comment by liuhy397** [comment]
> is it dram trace or fetch request generated at processor?

yes,it is a dram trace,do you know how to generate it ,thank you fot your help

**Comment by gangmul12** [comment]
almost every memory request has type name mem_fetch and vaiable name is usually mf. mem_fetch class has a method print that prints almost every detail.

mem ctrl is implemented in l2cache.cc so take a look some code in it( cycle is a good start point i think..) and put mf->print in the code

**Comment by liuhy397** [comment]
> almost every memory request has type name mem_fetch and vaiable name is usually mf. mem_fetch class has a method print that prints almost every detail.
> 
> mem ctrl is implemented in l2cache.cc so take a look some code in it( cycle is a good start point i think..) and put mf->print in the code

thank you very much ,i will have a try

**Comment by liuhy397** [comment]
> almost every memory request has type name mem_fetch and vaiable name is usually mf. mem_fetch class has a method print that prints almost every detail.
> 
> mem ctrl is implemented in l2cache.cc so take a look some code in it( cycle is a good start point i think..) and put mf->print in the code

dear gangmul12,I try it for several days,but I still do not know how to reduce a trace file,could you give me some help ,thank you ,goodman

**Comment by gangmul12** [comment]
What do you mean, reduce? Do you want tominizmize something?

**Comment by licaiwei397** [comment]
> What do you mean, reduce? Do you want tominizmize something?

sorry for my expression,i want to generate a trace file like the picture,thank you for help me,thank you very much

**Comment by gangmul12** [comment]
https://github.com/gpgpu-sim/gpgpu-sim_distribution/blob/7cd0edfe0cb654280c30afef89a563867d9e67ed/src/gpgpu-sim/l2cache.cc#L244-L250
how about put mf->print(stdout) before m_dram->push(mf)? i think it will work

**Comment by licaiwei397** [comment]
> https://github.com/gpgpu-sim/gpgpu-sim_distribution/blob/7cd0edfe0cb654280c30afef89a563867d9e67ed/src/gpgpu-sim/l2cache.cc#L244-L250
> 
> how about put mf->print(stdout) before m_dram->push(mf)? i think it will work

dear gangmul12,thank you again for your help,i follow your instruction and the print result is as below:
![mf](https://user-images.githubusercontent.com/39975865/62602955-7b006f80-b927-11e9-9b40-dfa80936455c.png)


what shoukd i do next in order to generate a trace file,thank you again ,it is my  most lucky to receive your help

**Comment by gangmul12** [comment]
@licaiwei397  do you mean the data from mf is not sufficient to generate a trace file?

**Comment by licaiwei397** [comment]
> @licaiwei397 do you mean the data from mf is not sufficient to generate a trace file?

yes,so what should i do to generate a trace file include cycle number etc,thank you again for your help

**Comment by gangmul12** [comment]
@licaiwei397 trace file is a general word, which does not specify any detail. is there any list of variables you want to include in the trace file?

**Comment by licaiwei397** [comment]
I want include the  cycle number ， memory request of DRAM and memory address corresponding to each cycle .Thank you .

**Comment by gangmul12** [comment]
addr is memory address, number in bracket() is cycle, INST is memory request type -- which mean it fetch instruction.. I think you just get everything from the print() function

**Comment by licaiwei397** [comment]
dear friend,i suffer a new question,i want to add a new mf request in l2 to dram queue,and print it in terminal,what code should i add,thank you for help me.
![image](https://user-images.githubusercontent.com/39975865/65506381-0052da00-defe-11e9-995d-b8c6e3ef97c2.png)

**Comment by licaiwei397** [comment]
> addr is memory address, number in bracket() is cycle, INST is memory request type -- which mean it fetch instruction.. I think you just get everything from the print() function
![10](https://user-images.githubusercontent.com/39975865/65571883-db5f7500-df98-11e9-956a-7dfd60d849da.png)
dear friend,i want to add a mf request,but i suffer a error when i write a code like this,please help me
![11](https://user-images.githubusercontent.com/39975865/65571942-15307b80-df99-11e9-959a-3ef1f5f269d9.png)

thank you for your selfless help

**Comment by licaiwei397** [comment]
> addr is memory address, number in bracket() is cycle, INST is memory request type -- which mean it fetch instruction.. I think you just get everything from the print() function
/home/licaiwei/Pictures/9.22/12.png
i amend it like this and suffer a abort,what should i do please help me,thank you
/home/licaiwei/Pictures/9.22/13.png

**Comment by licaiwei397** [comment]
> addr is memory address, number in bracket() is cycle, INST is memory request type -- which mean it fetch instruction.. I think you just get everything from the print() function

i write a mf like this and i suffer a new abort,what should i do ,thank you,thank you
![12](https://user-images.githubusercontent.com/39975865/65574271-92f78580-df9f-11e9-94f9-64eeb202dffd.png)
![13](https://user-images.githubusercontent.com/39975865/65574274-94c14900-df9f-11e9-8682-2bc4c8410539.png)

**Comment by gangmul12** [comment]
I think you need to debug where the error occurs. Use GDB and pinpoint where the source of the error is

**Comment by kamilkhan52** [comment]
![image](https://user-images.githubusercontent.com/45339822/75276776-0823ec80-57c4-11ea-9803-820c5bdca31c.png)

I made this change to the dram_sched.cc file
You may want to make sure if this is what you want.
